lib/Lift/RootAnalyzer.cpp

```diff
@@ -404,14 +404,14 @@ Function *RootAnalyzer::createTemporaryRoot(Function *TheFunction,
   {
     // Compute preserved registers using the default prototype
     using RegisterSet = llvm::SmallSet<model::Register::Values, 16>;
-    RegisterSet DefaultPreservedRegisters;
+    RegisterSet PreservedRegisters;
     model::ABI::Values ABI = Model->DefaultABI();
-    const auto &DefaultPrototype = Model->DefaultPrototype();
-    if (not DefaultPrototype.empty()) {
-      DefaultPreservedRegisters = getPreservedRegisters(DefaultPrototype);
+    if (not Model->DefaultPrototype().empty()) {
+      // TODO: flatten this check once `defaultPrototype` is always available.
```

Comment:
Context for this TODO?

Reply:
I do still feel like having "default prototype" encoded by the `Binary::DefaultABI` is the way to go. Especially since we definitely want to merge `abi::Definition` back into the model (the only way to get any meaningful CFT verification).

As far as I remember, your main concern about it was somebody abusing that and assigning those "temporary" prototypes to real functions, which would lead to unnecessary duplication we _definitely_ don't want.
I feel like we can work around that by providing "accessors" instead. Most of the places that use the prototype don't really care about the prototype itself, only some of its properties. And we already have some basic query for those after the recent `Layout` "gutting" - we'd still need to evaluate how feasible it is, _but_ IMO we never want to have non-user-defined `Binary::DefaultABI` in the model - there's no point of attaching the "default" prototype to a function - it can be implicit.

Either way, since this is a discussion for another day, I'm fine with dropping this TODO for now if it's bothering you.

Reply:
Having a `DefaultPrototype` is useful if we don't support a certain ABI, at least to get you started. For instance it'd be a solution in [this issue](https://github.com/revng/revng/issues/367).

If we enable defining new ABIs in the model, we might think about dropping `DefaultPrototype`. But that's kinda big change.

Anyway, I'm fine with the TODO, just rephrase as "consider doing X if we in the end decide of doing Y".

Reply:
Done:
```cpp
      // TODO: don't forget to simplify the logic here if we decide to make
      //       default prototypes always available (after merging
      //       `abi::Definition` back into the model).
```